dhcp-controller/api/v1alpha1/dhcpserver_types.go

```diff
@@ -48,13 +48,11 @@ type Server struct {
 type CIDR struct {
 	// refers to start IP of range
 	// +kubebuilder:validation:Required
+	CIDRIP string `json:"cidrIp,omitempty"`
+	// refers to start IP of range
 	RangeStartIp string `json:"rangeStartIp,omitempty"`
 	// refers to end IP of range
-	// +kubebuilder:validation:Required
 	RangeEndIp string `json:"rangeEndIp,omitempty"`
-	// refers to subnet mask
-	// +kubebuilder:validation:Required
-	RangeNetMask string `json:"rangeNetMask,omitempty"`
 	// refers to gateway IP
 	// +kubebuilder:validation:Required
 	GwAddress string `json:"gwAddress,omitempty"`
```

Comment:
Gateway must also be optional. Some networks dont have gateways. In fact, this can cause problems as 2ndary networks, if they have a GW advertised via DHCP cause the host to overwrite the default route added by the 1st network. What this means is the last network always attached to a VM/Pod will have the default route.

For this very reason in DF neutron, we have the gateway disabled on our Multus networks: https://df.platform9.net/clarity/index.html#/networks/details/86564a47-9c24-472b-aeb2-81bd8dc9e318#subnets

The reason being we want default route to always go out primary network (the 1st NIC, used for Calico network/Floating IP on host), we dont want the hosts to have the default route going over the networks used for macvlan/OVS

Reply:
I see its omitempty here (but so is the cidr, which should not be), but making sure it works if unset - we won't want to set the dhcp option in dnsmasq to advertise any GW.

Reply:
Done